sebs/knative/trigger.py

```diff
@@ -0,0 +1,295 @@
+import concurrent.futures
+import datetime
+import json
+import requests
+import subprocess
+import time
+from typing import Dict, List, Optional  # noqa
+
+from sebs.faas.function import ExecutionResult, Trigger
+
+
+class LibraryTrigger(Trigger):
+    """
+    Trigger implementation for invoking a Knative service using port forwarding and curl.
+
+    Attributes:
+        function_name (str): The name of the function to invoke.
+        pod_name (str): The name of the Kubernetes pod where the function is deployed.
+        namespace (str): The Kubernetes namespace where the pod is deployed (default is 'default').
+    """
+
+    def __init__(self, function_name: str, pod_name: str, namespace: str = "default"):
+        """
+        Initialize the LibraryTrigger with the function name, pod name, and namespace.
+
+        Args:
+            function_name (str): The name of the function to invoke.
+            pod_name (str): The name of the Kubernetes pod where the function is deployed.
+            namespace (str, optional): The Kubernetes namespace where the pod is deployed (default is 'default').
+        """
+        super().__init__()
+        self.function_name = function_name
+        self.pod_name = pod_name
+        self.namespace = namespace
+
+    @staticmethod
+    def trigger_type() -> "Trigger.TriggerType":
+        """
+        Return the type of trigger (LibraryTrigger).
+
+        Returns:
+            Trigger.TriggerType: The trigger type (LibraryTrigger).
+        """
+        return Trigger.TriggerType.LIBRARY
+
+    @staticmethod
+    def get_curl_command(payload: dict) -> List[str]:
+        """
+        Generate a curl command for invoking the function.
+
+        Args:
+            payload (dict): The payload data to send with the request.
+
+        Returns:
+            List[str]: The curl command as a list of strings.
+        """
+        return [
+            "curl",
+            "-X",
+            "POST",
+            "http://localhost:8080/handle",
```

Comment:
We shouldn't be hardcoding any addresses or ports.

Reply:
I updated the implementation, reason:https://github.com/spcl/serverless-benchmarks/pull/206#discussion_r1661859729